app/controllers/concerns/cohort_period_selection.rb

```diff
@@ -28,7 +23,7 @@ def sanitize_registration_quarter(quarter_param)
 
     def sanitize_registration_year(year)
       return year.to_i if year.to_i.positive?
-      return (Time.current.beginning_of_month - 1.month).year if @period == :month
+      return (Time.current.beginning_of_month - 1.month).year if @selected_cohort_period == :month
```

Comment:
I think there is still a bug here. @selected_cohort_period is the whole hash, not a single symbol like `:month`